src/main/java/org/cryptomator/common/PropertiesPreprocessor.java

```diff
@@ -0,0 +1,66 @@
+package org.cryptomator.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class PropertiesPreprocessor {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PropertiesPreprocessor.class);
+	private static final Pattern TEMPLATE = Pattern.compile("@\\((.+)\\)");
```

Comment:
This Pattern is greedy, it would not only match `@(hello)` but also `@(hello) world)`. To be 100% exact, we would have to add an escape mechanism (`@(hello\) world)`), which would probably not work with regex but requires a manually written state machine.

To keep it simple, we might want to use a regex which disallows certain characters (e.g. `@\\(([^)]+)\\)`), or even simpler just allows certain chars (`@\\((\w)\\)`).

---

On a separate note: Maybe we should not use `@()` but rather `@{}`.  `{}` don't need to be quoted, making the regex more legible, furthermore these characters are less common in the actual property (maybe, we want to eventually support `some.property.with.@{thing.getName()}`)

Reply:
Only allow certain characters is a good idea!

Regarding the token: Actually, `{` and `}` must be quoted as well because they are used in character classes and quantifiers. But the latter part about calling functions is a good point!